src/client/datascience/jupyter/kernels/kernelSelector.ts

```diff
@@ -147,8 +147,7 @@ export class KernelSelector implements IKernelSelectionUsage {
         notebookMetadata?: nbformat.INotebookMetadata,
         disableUI?: boolean,
         cancelToken?: CancellationToken,
-        ignoreDependencyCheck?: boolean,
-        ignoreTrackingKernelInformation?: boolean
```

Comment:
Removed the hacks & assumptions.
Just because some of these methods were called doesn't mean the kernel was changed.
We now track the kernel against the URI where these methods are called & kernel information is actually changed/set. No assumptions.